app/src/main/java/com/guruthedev/instagram/SharedPreference.kt

```diff
@@ -0,0 +1,65 @@
+package com.guruthedev.instagram
+
+import android.content.Context
+import android.content.Intent
+import android.content.SharedPreferences
+import com.guruthedev.instagram.ui.fragments.auth.login.LoginFragment
+import com.guruthedev.instagram.utils.Constants.IS_LOGIN
+import com.guruthedev.instagram.utils.Constants.KEY_EMAIL
+import com.guruthedev.instagram.utils.Constants.KEY_FULL_NAME
+import com.guruthedev.instagram.utils.Constants.KEY_USERNAME
+import com.guruthedev.instagram.utils.Constants.PREF_NAME
+
+class SharedPreference {
+    private lateinit var preference: SharedPreferences
+    private lateinit var editor: SharedPreferences.Editor
+    lateinit var context: Context
+
+    constructor(context: Context) {
+        this.context = context
+        preference = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE)
+        editor = preference.edit()
+    }
+
+    fun createLoginSession(fullName: String, username: String, email: String) {
+        editor.apply {
+            putBoolean(IS_LOGIN, true)
+            putString(KEY_FULL_NAME, fullName)
+            putString(KEY_USERNAME, username)
+            putString(KEY_EMAIL, email)
+            commit()
+        }
+    }
+
+    fun checkLogin() {
+        if (!this.isLoggedIn()) {
+            val intent: Intent = Intent(context, LoginFragment::class.java)
```

Comment:
Will this code work ?
as you are trying to Navigate to LoginFragment which is a Fragment not Activity

Reply:
no this code not working